### PR TITLE
Commenting take 27

### PR DESCRIPTION
**Pull request by RobertWDLowe**

### CODE REVIEW

File: wp-content/themes/mojintranet/inc/comments.php [hunk 1]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
...
+  if(in_array($pagenow, array('post-new.php')) && $post_type == 'post' ) {
+    $comments_on = 1;
+  }
+  else {
+    $comments_on = get_post_meta($post->ID, 'dw_comments_on', true);
+  }
+  ?>
+  <input name="advanced_view" type="hidden" value="1" />
+  <p class="meta-options">
+    <label for="comments_on" class="selectit"><input name="comments_on" type="checkbox" id="comments_on" value="1"  <?php checked($comments_on, 1); ?> /> <?php _e('Comments on') ?></label>
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/inc/comments.php]: @RobertWDLowe Suggested new wording?
"Show the comments section"
"Allow people to add new comments"

File: wp-content/themes/mojintranet/inc/comments.php [hunk 2]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
...
+    do_action( 'post_comment_status_meta_box-options', $post );
+    ?>
+  </p>
+  <?php
+}
+
+function dw_set_comments_on_meta($post_id) {
+  $post_type = get_post_type($post_id);
+
+
```
Comment (RobertWDLowe) [inline wp-content/themes/mojintranet/inc/comments.php]: Lose space

File: wp-content/themes/mojintranet/inc/comments.php [hunk 3]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
...
+  if ($hook == 'post-new.php' || $hook == 'post.php') {
+    if (in_array($post->post_type, ['post', 'page'])) {
+      wp_enqueue_script('discussion-options', get_stylesheet_directory_uri().'/admin/js/discussion-options.js');
+    }
+  }
+}
+add_action('admin_enqueue_scripts', 'dw_discussion_options_script', 10, 1);
+
+function remove_duplicate_check($dupe_id, $commentdata) {
+  return false;
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/inc/comments.php]: @RobertWDLowe replace `remove_duplicate_check` method with filter callback `__return_false`.

File: wp-content/themes/mojintranet/inc/comments.php [hunk 4]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
 
   new EnhanceComments;
 
-?>
\ No newline at end of file
+
+function dw_add_new_discussion_meta_box() {
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/inc/comments.php]: @RobertWDLowe Please add docblocks for the new methods you've added to these files. And **rename** the `EnhanceComments` class to something more that better describes that class's function.

File: wp-content/themes/mojintranet/inc/comments.php [hunk 5]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
...
+    update_metadata('post', $post_id, 'dw_comments_on', 0);
+  }
+}
+add_action('save_post', 'dw_set_comments_on_meta', 10, 1);
+
+function dw_save_comment_status($data , $postarr) {
+  global  $post;
+
+  if(!empty($_POST["comment_status"])) {
+
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/inc/comments.php]: @RobertWDLowe please fix spacing issues

File: wp-content/themes/mojintranet/inc/comments.php [hunk 6]
```diff
@@ -90,4 +92,132 @@ public function display_comment_hidden_column($column, $comment_id) {
...
+          $comment_data['comment_meta']['root_comment_id'] = $root_comment_id;
+      }
+  }
+
+  return $comment_data;
+}
+add_filter('preprocess_comment','dw_add_root_comment',10,1);
+
+
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/inc/comments.php]: @RobertWDLowe Please remove extra spaces from end of file. Thx.

File: wp-content/themes/mojintranet/single-post.php [hunk 7]
```diff
@@ -19,20 +19,37 @@ function get_data(){
...
+      $revisions = wp_get_post_revisions($post->ID);
+
+      if (count($revisions) > 0) {
+        $latest_revision = array_shift($revisions);
+        $this_id = $latest_revision->ID;
+      }
+
+    }
+
+
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/single-post.php]: @RobertWDLowe Please remove whitespace 😵

File: wp-content/themes/mojintranet/views/modules/request_activation_link/form.php [hunk 8]
```diff
@@ -0,0 +1,37 @@
+<?php if (!defined('ABSPATH')) die(); ?>
+
+<div class="grid register-form-box not-logged-in-only">
+  <div class="col-lg-8 col-md-12 col-sm-12">
+    <h3>Comment on this page</h3>
+
+    <form class="userform standard register-form">
+      <p class="description">Fill in your details below. We’ll then send you a link back to this page so you can start commenting.</p>
```
Comment (ollietreend) [inline wp-content/themes/mojintranet/views/modules/request_activation_link/form.php]: Suggested rewording to remove "then":

Fill in your details below. We’ll email you a link back to this page so you can start commenting.

**Pull request merged**